spyder/plugins/switcher/widgets/switcher.py

```diff
@@ -311,6 +338,8 @@ def setup(self):
                 rich_title = rich_title.replace(" ", "&nbsp;")
                 item.set_rich_title(rich_title)
             item.set_score(score_value)
+
+        self.sig_search_text_available.emit(search_text)
```

Comment:
Following the comments above, this signal could emit not only the `search_text` but also something like a new list variable that could be called `items_data` which could constructed while looping the model